x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_selector/index.tsx

```diff
@@ -96,61 +100,68 @@ export const ConversationSelector: React.FC<Props> = React.memo(
               option.label.trim().toLowerCase() === normalizedSearchValue
           ) !== -1;
 
+        let cId;
         if (!optionExists) {
           const newConversation: Conversation = {
             id: searchValue,
+            title: searchValue,
             messages: [],
             apiConfig: {
               connectorId: defaultConnectorId,
               provider: defaultProvider,
               defaultSystemPromptId: defaultSystemPrompt?.id,
             },
           };
-          setConversation({ conversation: newConversation });
+          cId = (await createConversation(newConversation))?.id;
         }
-        onConversationSelected(searchValue);
+        onConversationSelected(cId ?? DEFAULT_CONVERSATION_TITLE);
       },
       [
         allSystemPrompts,
+        onConversationSelected,
         defaultConnectorId,
         defaultProvider,
-        setConversation,
-        onConversationSelected,
+        createConversation,
       ]
     );
 
     // Callback for when user deletes a conversation
     const onDelete = useCallback(
       (cId: string) => {
+        onConversationDeleted(cId);
         if (selectedConversationId === cId) {
-          onConversationSelected(getPreviousConversationId(conversationIds, cId));
+          const prevConversationId = getPreviousConversationId(conversationIds, cId);
+          onConversationSelected(prevConversationId, conversations[prevConversationId].title);
         }
-        setTimeout(() => {
-          deleteConversation(cId);
-        }, 0);
       },
-      [conversationIds, deleteConversation, selectedConversationId, onConversationSelected]
+      [
+        selectedConversationId,
+        onConversationDeleted,
+        onConversationSelected,
+        conversationIds,
+        conversations,
+      ]
     );
 
     const onChange = useCallback(
-      (newOptions: ConversationSelectorOption[]) => {
-        if (newOptions.length === 0) {
+      async (newOptions: ConversationSelectorOption[]) => {
+        if (newOptions.length === 0 || !newOptions?.[0].id) {
           setSelectedOptions([]);
-        } else if (conversationOptions.findIndex((o) => o.label === newOptions?.[0].label) !== -1) {
-          onConversationSelected(newOptions?.[0].label);
+        } else if (conversationOptions.findIndex((o) => o.id === newOptions?.[0].id) !== -1) {
+          await onConversationSelected(newOptions?.[0].id, newOptions?.[0].label);
```

Comment:
To avoid repeating

```ts
newOptions?.[0]
```

 on `148` - `151`:

consider something like

```ts
const { id, label } = newOptions?.[0];
```